relay-filter/src/generic_filters.rs

```diff
@@ -0,0 +1,98 @@
+//! Implements generic filtering based on the [`RuleCondition`] DSL.
+//!
+//! Multiple generic filters can be defined and they are going to be checked in FIFO order. The
+//! first one that matches, will result in the event being discarded with a [`FilterStatKey`]
+//! identifying the matching filter.
+
+use crate::{FilterStatKey, GenericFiltersConfig};
+use relay_event_schema::protocol::Event;
+use relay_sampling::condition::RuleCondition;
+
+/// Checks events by patterns in their error messages.
+pub fn matches(event: &Event, condition: Option<&RuleCondition>) -> bool {
+    // TODO: the condition DSL needs to be extended to support more complex semantics, such as
+    //  collections operations.
+    condition.map_or(false, |condition| condition.matches(event))
+}
+
+/// Filters events by patterns in their error messages.
+pub fn should_filter(event: &Event, config: &GenericFiltersConfig) -> Result<(), FilterStatKey> {
+    for (filter_name, filter_config) in config.0.iter() {
+        if filter_config.is_enabled && matches(event, filter_config.condition.as_ref()) {
+            return Err(FilterStatKey::GenericFilter(filter_name.clone()));
+        }
+    }
+
+    Ok(())
+}
+
+#[cfg(test)]
+mod tests {
+    use crate::generic_filters::should_filter;
+    use crate::{FilterStatKey, GenericFilterConfig, GenericFiltersConfig};
+    use relay_event_schema::protocol::{Event, LenientString};
+    use relay_protocol::Annotated;
+    use relay_sampling::condition::RuleCondition;
+
+    fn mock_filters() -> Vec<(String, GenericFilterConfig)> {
+        vec![
+            (
+                "hydrationError".to_string(),
+                GenericFilterConfig {
+                    is_enabled: true,
+                    condition: Some(RuleCondition::eq("event.release", "1.0")),
```

Comment:
Could we use a filter name that better reflects the condition and avoid some confusion? Same for the filter below.